MueLu:  Fails to compile c++17 due to std::ptr_fun

## Bug Report
@trilinos/MueLu

### Description
MueLu is not compiling with apple-clang when I turn on c++17.
```terminal
     2028    In file included from /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp:57:
  >> 2029    /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp:131:74: error: no member named 'ptr_fun' in namespace 'std'
     2030          s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
     2031                                                                        ~~~~~^
  >> 2032    /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp:131:85: error: expected '(' for function-style cast or type construction
     2033          s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
     2034                                                                                     ~~~^
  >> 2035    /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp:131:90: error: expected '(' for function-style cast or type construction
     2036          s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
     2037                                                                                          ~~~^
  >> 2038    /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp:137:65: error: no member named 'ptr_fun' in namespace 'std'
     2039          s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
     2040                                                               ~~~~~^
  >> 2041    /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp:137:76: error: expected '(' for function-style cast or type construction
     2042          s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
     2043                                                                            ~~~^
  >> 2044    /Users/psakiev/soft/spack-manager/spack/var/spack/stage/spack-stage-trilinos-develop-37mbkz4mczqp2ju2rck4cwnk3vuzew25/spack-src/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp:137:81: error: expected '(' for function-style cast or type construction
     2045          s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
     2046                                                                                 ~~~^
     2047    6 errors generated.
```

### Steps to reproduce
Compile MueLu from the develop branch with clang/apple-clang and c++17

--- a/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp
+++ b/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.hpp
@@ -128,13 +128,13 @@ namespace MueLu {
 
     // trim from start
     static inline std::string &ltrim(std::string &s) {
-      s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+      s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int c){return !std::isspace(c);}));
       return s;
     }
 
     // trim from end
     static inline std::string &rtrim(std::string &s) {
-      s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
+      s.erase(std::find_if(s.rbegin(), s.rend(), [](int c){return !std::isspace(c);}).base(), s.end());
       return s;
     }
 

--- a/packages/seacas/applications/ejoin/EJ_SystemInterface.h
+++ b/packages/seacas/applications/ejoin/EJ_SystemInterface.h
@@ -98,7 +98,7 @@ private:
   bool zlib_{true};
   bool szip_{false};
 
-  std::string blockPrefix_{std::string("p")};
+  std::string blockPrefix_{"p"};
 
   vector3d offset_;
   double   tolerance_{0.0};

--- a/packages/tpetra/core/src/Tpetra_Import_def.hpp
+++ b/packages/tpetra/core/src/Tpetra_Import_def.hpp
@@ -1177,7 +1177,7 @@ namespace Tpetra {
 
       const size_type numInvalidRemote =
         std::count_if (remoteProcIDs.begin (), remoteProcIDs.end (),
-                       std::bind1st (std::equal_to<int> (), -1));
+                       std::bind (std::equal_to<int> (), -1, std::placeholders::_1));
       TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
         (numInvalidRemote == 0, std::logic_error, "Calling getRemoteIndexList "
          "on the source Map returned IDNotPresent, but none of the returned "

